python/ml4ir/base/model/scoring/monte_carlo_scorer.py

```diff
@@ -120,16 +118,19 @@ def call(self, inputs: Dict[str, tf.Tensor], training=None):
         scores : dict of tensor object
             Tensor object of the score computed by the model
         """
-        # TODO replace loop with vector operations for performance
         if training:
-            monte_carlo_trials = self.monte_carlo_training_trials
-            monte_carlo_trials_tf = self.monte_carlo_training_trials_tf
+            if not self.use_fixed_mask_in_training:
+                monte_carlo_trials = self.monte_carlo_training_trials
+            else:
+                monte_carlo_trials = QueryFeatureMask.masking_config[MonteCarloInferenceKey.FIXED_MASK_COUNT] - 1
         else:
-            monte_carlo_trials = self.monte_carlo_test_trials
-            monte_carlo_trials_tf = self.monte_carlo_test_trials_tf
+            if not self.use_fixed_mask_in_testing:
+                monte_carlo_trials = self.monte_carlo_test_trials
+            else:
+                monte_carlo_trials = QueryFeatureMask.masking_config[MonteCarloInferenceKey.FIXED_MASK_COUNT] - 1
```

Comment:
@mohazahran why can't this be done in the constructor? It's an unnecessary if-else ladder on each call, no?

Reply:
Yes, I understand why you do it this way. I was wondering if we will use all those different train vs test combinations in practice. Should be fine for now.

Reply:
The current model we're interested in actually does that. It uses stochastic MC in training and we will use deterministic MC in testing.